move/sources/refund.move

```diff
@@ -0,0 +1,429 @@
+    public fun current_liabilities(pool: &RefundPool): u64 { accounting::current_liabilities(&pool.accounting) }
+
+    // === Friends ===
+
+    public(friend) fun uid_mut(pool: &mut RefundPool): &mut UID { &mut pool.id }
+
+    public(friend) fun unclaimed_mut(pool: &mut RefundPool): &mut Table<address, u64> { &mut pool.unclaimed }
+    public(friend) fun accounting_mut(pool: &mut RefundPool): &mut Accounting { &mut pool.accounting }
+    public(friend) fun booster_pool_mut(pool: &mut RefundPool): &mut Pool { &mut pool.base_pool }
+
+    public(friend) fun reclaim_funds_(
+        inner_pool: &mut Pool,
+        total_raised: u64,
+        ctx: &mut TxContext,
+    ) {
+        let funders = funders_mut(inner_pool);
+        assert!(table::contains(funders, sender(ctx)), EInvalidFunder);
+        let funding_amount = table::remove(funders, sender(ctx));
+
+        let is_last = table::is_empty(funders);
+        let funds = funds_mut(inner_pool);
+        
+        let reclaim_amount = if (is_last) {
+            balance::value(funds)
+        } else {
+            let leftovers = balance::value(funds);
+
+            // ReclaimAmount = Leftovers * % Share <=>
+            // ReclaimAmount = Leftovers * FundingAmount/TotalRaised
+            // 
+            // We first upscale then downscale
+            div(
+                mul(leftovers, funding_amount),
+                total_raised
+            )
+        };
+
+        let reclaim_funds = coin::from_balance(balance::split(funds, reclaim_amount), ctx);
+        transfer::public_transfer(reclaim_funds, sender(ctx));
+    }
+
+    // === Phase Transitions ===
+    
+    public entry fun start_funding_phase(
+        pub: &Publisher,
+        pool: &mut RefundPool,
+        timeout_ts: u64,
+        clock: &Clock,
+    ) {
+        assert_publisher(pub);
+        assert_address_addition_phase(pool);
+        
+        assert!(timeout_ts > clock::timestamp_ms(clock), EInvalidTimeoutTimestamp);
+        option::fill(&mut pool.timeout_ts, timeout_ts);
+
+        next_phase(pool)
+    }
+
+    public entry fun start_claim_phase(
+        pool: &mut RefundPool,
+    ) {
+        assert_funding_phase(pool);
+        let total_to_refund = total_to_refund(&pool.accounting);
+        let total_raised = total_raised(&pool.accounting);
+        assert!(total_to_refund == total_raised, EPoolUnderfunded);
+
+        next_phase(pool)
+    }
+    
+    public entry fun start_reclaim_phase(
+        pool: &mut RefundPool,
+        clock: &Clock,
+    ) {
+        let timeout_ts = option::borrow(&pool.timeout_ts);
+        assert!(clock::timestamp_ms(clock) >= *timeout_ts, ECurrentTimeNotAboveTimeout);
+
+        if (is_funding_phase(pool)) {
+            let total_to_refund = total_to_refund(&pool.accounting);
+            let total_raised = total_raised(&pool.accounting);
+            assert!(total_raised < total_to_refund, 0);
+            
```

Comment:
**I think this can become a vulnerability, and should be removed**. 
"Funders" can get their funds back calling `withdraw_funds` in the funding phase.

If you call this function once (and not twice), you would effectively unblock user claims. (there's no check on the `timeout_ms` when claiming funds).

Reply:
`withdraw_funds` has been removed now